Source/HelixToolkit.Wpf.SharpDX/Controls/ViewportExtensions.cs

```diff
@@ -355,12 +355,20 @@ public static Ray UnProject(this Viewport3DX viewport, Vector2 point2d)//, out V
 
                 var vp = GetScreenViewProjectionMatrix(viewport);
                 var vpi = Matrix.Invert(vp);
+
+                var test = 1f / ((p.X * vpi.M14) + (p.Y * vpi.M24) + (p.Z * vpi.M34) + vpi.M44);

```

Comment:
This check was added to fix an instability in Pan operations when the far clip plane was very large.